README.md

```diff
@@ -55,6 +55,33 @@ Documentation will be generated in target/doc/...
 Most documentation is taken from NaCl, with minor modification where the API
 differs between the C and Rust versions.
 
+#### Building and Testing with Support for AES
+
+The AES AEAD variant `crypto_aead_aes256gcm` requires hardware support for the
+`AES` and `CLMUL` instruction set extensions to x86; you can read why that's the
+case
+[here](https://libsodium.gitbook.io/doc/secret-key_cryptography/aead/aes-256-gcm#limitations). These instruction set extensions were first made
+available in Intel Westmere (early 2010) and at the time of writing x86 hardware
+support for them is near universal. However, rustc is a cross compiler and will
+not enable them unless it's targeting a CPU that supports them; the baseline
+x86 target does not.
+
+If you need to use `crypto_aead_aes256gcm` you must build/test sodiumoxide with
+`RUSTFLAGS` set to include one of the following option sets:
```

Comment:
I would suggest avoid advocating use of the `RUSTFLAGS` enviroment variable here as in a production setting this should often be done via [Cargo configuration](https://doc.rust-lang.org/cargo/reference/config.html) instead. Maybe talk about `rustc` flags in a generic manner?